### ISSUE TITLE
[Keybase] Implement an encrypted wrapper for private keys akin to `ArmoredJson` from V0

### ISSUE THREAD

**Issue by deblasis** [pokt-network/pocket#184]
\__DELETE_ME_: Please make sure to update the _Assigness_, _Labels_, _Projects_, _Milestone_ before submitting the issue\_
\__DELETE_ME_: Please make sure to add labels for the purpose (e.g. `persistence`) and priority (e.g. `low`) of this PR\_

## Objective

Implement a secure wrapper around private keys.

## Origin Document

For increased security, we are going to need something like `ArmoredJson` from [v0](https://github.com/pokt-network/pocket-core/blob/08df7217690a89156ae1e7bd54505be3dfd2dfc2/crypto/keys/mintkey/mintkey.go#L96) to protecting Private keys coming out of a keybase and providing a way of importing new private keys into the keybase.

This issue is dependent on #150 but I believe it's still possible to work on this in parallel by developing the core functionality for encryption/decryption for initial use with the Utility CLI (#112) where, for now, the privatekeys come into play in the form of a clear-text JSON file. 
We just have to remember (via TODOs and/or GH issues) that ultimately the keybase should be interfaced with this as well.

## Goals

- [ ] Determine if the V0 `ArmoredJson` implementation could be used in V1 as well and if not, propose alternatives with pros/cons comparisons
- [ ] Define interfaces and methods to provide access to the underlying private keys given an address and a passphrase
- [ ] Define interfaces and methods to allow importing of private keys into the keybase
- [ ] Use these new interfaces/methods to access private keys where needed (a good place to start is the CLI, starting from the Utility commands tracked in #112 )
- [ ] Create [import](https://github.com/pokt-network/pocket-core/blob/08df7217690a89156ae1e7bd54505be3dfd2dfc2/app/cmd/cli/accounts.go#L325) functionality for private keys
- [ ] Create [export](https://github.com/pokt-network/pocket-core/blob/08df7217690a89156ae1e7bd54505be3dfd2dfc2/app/cmd/cli/accounts.go#L359) functionality for private keys

## Deliverable

- [ ] ArmoredJson / ArmoredJson-like implementation with basic `Encrypt` and `Decrypt` methods (see [v0 ArmoredJson](https://github.com/pokt-network/pocket-core/blob/08df7217690a89156ae1e7bd54505be3dfd2dfc2/crypto/keys/mintkey/mintkey.go#L96))
- [ ] Import account from encrypted and ASCII armored file CLI command
- [ ] Export account from encrypted and ASCII armored file CLI command
- [ ] Log any unfinished work with TODOs depending on the state of #150 and #112 at the moment this is completed/ready to merge.

## Non-goals / Non-deliverables

- Do any work currently tracked by other Issues (especially #150 and #112)

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)
- Write tests that ensure encryption and decryption work as expected, including KO cases (eg: wrong passphrase)

---

**Creator**: @deblasis 
**Co-Owners**: @Olshansk, @andrewnguyen22

**Comment by Olshansk** [comment]
@h5law completed this as part of the key base work.
